File: SPCode/CS/Security/Support/UserManagerProxy.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
using System.Web.Services.Protocols;
using System.Web.Services;
using System.Configuration;


namespace SoftLogik.Win
{
	namespace Security
	{
		[DebuggerStepThrough(), WebServiceBinding(Name = "WindowsUserManagerSoap", Namespace = "http://SecurityServices")]partial class UserManager : SoapHttpClientProtocol, IUserManager
		{
			
			public UserManager()
			{
				CookieContainer = new System.Net.CookieContainer();
				string urlSetting = ConfigurationManager.AppSettings["UserManager"];
				if (urlSetting != null)
				{
					Url = urlSetting;
				}
				else
				{
					Trace.TraceWarning("No URL was found in application configuration file");
				}
			}
			public UserManager(string url)
			{
				CookieContainer = new System.Net.CookieContainer();
				url = url;
			}
			[SoapDocumentMethod("http://SecurityServices/Authenticate")]public bool Authenticate(string applicationName, string userName, string password)
			{
				string[] parameters = {applicationName, userName, password};
				object[] results = Invoke("Authenticate", parameters);
				return (System.Convert.ToBoolean(results[0]));
			}
			[SoapDocumentMethod("http://SecurityServices/IsInRole")]public bool IsInRole(string applicationName, string userName, string role)
			{
				string[] parameters = {applicationName, userName, role};
				object[] results = Invoke("IsInRole", parameters);
				return (System.Convert.ToBoolean(results[0]));
			}
			[SoapDocumentMethod("http://SecurityServices/GetRoles")]public string[] GetRoles(string applicationName, string userName)
			{
				string[] parameters = {applicationName, userName};
				object[] results = Invoke("GetRoles", parameters);
				return ((string[]) (results[0]));
			}
		}
	}
	
}

File: SPCode/Security/Support/UserManagerProxy.vb
Imports System
Imports System.Web.Services.Protocols
Imports System.Web.Services
Imports System.Diagnostics
Imports System.Configuration

Namespace Security
    <DebuggerStepThrough(), WebServiceBinding(Name:="WindowsUserManagerSoap", Namespace:="http://SecurityServices")> _
    Partial Friend Class UserManager
        Inherits SoapHttpClientProtocol
        Implements IUserManager
        Public Sub New()
            CookieContainer = New System.Net.CookieContainer()
            Dim urlSetting As String = ConfigurationManager.AppSettings("UserManager")
            If Not urlSetting Is Nothing Then
                Url = urlSetting
            Else
                Trace.TraceWarning("No URL was found in application configuration file")
            End If
        End Sub
        Public Sub New(ByVal url As String)
            CookieContainer = New System.Net.CookieContainer()
            url = url
        End Sub
        <SoapDocumentMethod("http://SecurityServices/Authenticate")> _
        Public Function Authenticate(ByVal applicationName As String, ByVal userName As String, ByVal password As String) As Boolean Implements Security.IUserManager.Authenticate
            Dim parameters As String() = {applicationName, userName, password}
            Dim results As Object() = Invoke("Authenticate", parameters)
            Return (CBool(results(0)))
        End Function
        <SoapDocumentMethod("http://SecurityServices/IsInRole")> _
        Public Function IsInRole(ByVal applicationName As String, ByVal userName As String, ByVal role As String) As Boolean Implements Security.IUserManager.IsInRole
            Dim parameters As String() = {applicationName, userName, role}
            Dim results As Object() = Invoke("IsInRole", parameters)
            Return (CBool(results(0)))
        End Function
        <SoapDocumentMethod("http://SecurityServices/GetRoles")> _
        Public Function GetRoles(ByVal applicationName As String, ByVal userName As String) As String() Implements Security.IUserManager.GetRoles
            Dim parameters As String() = {applicationName, userName}
            Dim results As Object() = Invoke("GetRoles", parameters)
            Return (CType(results(0), String()))
        End Function
    End Class
End Namespace
